src/ui/RealmBrowser/Content/CreateObjectDialog/types/ObjectIdControl.tsx

```diff
@@ -0,0 +1,102 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2020 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import React from 'react';
+import { ObjectId } from 'bson';
+import { Button, Input, InputGroup, InputGroupAddon } from 'reactstrap';
+
+import { IBaseControlProps } from './TypeControl';
+import { parseObjectId } from '../../../parsers';
+
+interface IObjectIdControlState {
+  internalValue: string | null;
+}
+
+export class ObjectIdControl extends React.PureComponent<
+  IBaseControlProps<ObjectId | null>,
+  IObjectIdControlState
+> {
+  state: IObjectIdControlState = {
+    internalValue: this.props.value?.toHexString() ?? null,
```

Comment:
Funky ... I think it's the first time this codebase sees both the optional chaining and the nullish coalescing operator.
... but, there's a first time for everything :) It's less repeated and removes the edge-case of internalValue becomming null in the case of .toHexString() returning the empty string (if that can ever happen?).

Reply:
The nullish coalescing operator is just to convert the potential `undefined` value returned from optional chaining, back into `null` (if `value` is `null`).
If `toHexString` returns an empty string (don't know if it can either...), then the empty string would be used, `??` only checks for `undefined` & `null`, otherwise we should use the "good" old `||`. But the control should handle both `ObjectId` & `null` in the case of an optional property.